src/Compiler.Dynamic/DynamicControlCollection.cs

```diff
@@ -14,39 +17,41 @@ internal sealed class DynamicControlCollection : ITypeResolutionService, IMetada
 {
     private readonly AssemblyLoadContext _context;
     private readonly ILogger<DynamicControlCollection> _logger;
-    private readonly Lazy<LoadedAssemblies> _loaded;
+    private readonly LoadedAssemblies _loader;
 
     public DynamicControlCollection(ILogger<DynamicControlCollection> logger)
     {
         _logger = logger;
         _context = AssemblyLoadContext.Default;
+        _loader = new LoadedAssemblies(_logger);
 
-        _loaded = new(() =>
-        {
-            var loader = new LoadedAssemblies(_logger);
+        Initialize();
+    }
 
-            AppDomain.CurrentDomain.AssemblyLoad += CurrentDomain_AssemblyLoad;
+    private void Initialize()
```

Comment:
I couldn't figure out how to lazily load them since we need to actually load them into the current context without cycles... so I removed the lazily initialization